Tests/Realm.Tests/Database/AsyncTests.cs

```diff
@@ -155,5 +158,405 @@ static async Task<long> MeasureTiming(Func<Task> func)
                 }
             });
         }
+
+        [Test]
+        public void AsyncWrite_Action()
+        {
+            TestHelpers.RunAsyncTest(async () =>
+            {
+                var peopleQuery = _realm.All<Person>();
+                var personName = "Jonh Wickie";
+                Assert.That(peopleQuery.Count(), Is.EqualTo(0));
+
+                await _realm.WriteAsync(() =>
+                {
+                    var person = new Person { FullName = personName };
+                    _realm.Add(person);
+                });
+                Assert.That(peopleQuery.Count(), Is.EqualTo(1));
+                Assert.That(peopleQuery.First().FullName, Is.EqualTo(personName));
+            });
+        }
+
+        [Test]
+        public void AsyncBeginWriteAndCommit()
+        {
+            TestHelpers.RunAsyncTest(async () =>
+            {
+                var peopleQuery = _realm.All<Person>();
+                var personName = "Jonh Wickie";
+                Assert.That(peopleQuery.Count(), Is.EqualTo(0));
+
+                var transaction = await _realm.BeginWriteAsync();
+                var person = new Person { FullName = personName };
+                _realm.Add(person);
+                await transaction.CommitAsync();
+
+                Assert.That(peopleQuery.Count(), Is.EqualTo(1));
+                Assert.That(peopleQuery.First().FullName, Is.EqualTo(personName));
+            });
+        }
+
+        [Test]
+        public void AsyncBeginWrite_RollBack_DoesNot_PersistData()
+        {
+            TestHelpers.RunAsyncTest(async () =>
+            {
+                var personName = "Jonh Wickie";
+
+                using (var transaction = await _realm.BeginWriteAsync())
+                {
+                    var person = new Person { FullName = personName };
+                    _realm.Add(person);
+                    transaction.Rollback();
+                }
+
+                Assert.That(_realm.All<Person>().Count(), Is.EqualTo(0));
+            });
+        }
+
+        [Test]
+        public void AsyncBeginWriteAndClose_DoesNot_PersistData()
+        {
+            TestHelpers.RunAsyncTest(async () =>
+            {
+                IQueryable<Person> peopleQuery = null;
+
+                using (var innerRealm = GetRealm(_realm.Config))
```

Comment:
Why do we need an extra Realm instance for this test?

Reply:
But in the test you have the transaction inside a `using` statement - so before the Realm is closed, the transaction is going to get disposed and thus rolled back.

Reply:
Marked `Explicit` for the future